sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+
+                        object_id = asyncio.run(get_object_id())
+                        # Defintion ID of Storage Blob Data Owner role.
+                        # Reference: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-owner # pylint: disable=line-too-long
+                        storage_blob_data_owner_role_id = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
+                        role_definition_id = (
+                            '/subscriptions'
+                            f'/{azure.get_subscription_id()}'
+                            '/providers/Microsoft.Authorization'
+                            '/roleDefinitions'
+                            f'/{storage_blob_data_owner_role_id}')
+                        authorization_client.role_assignments.create(
+                            scope=creation_response.id,
+                            role_assignment_name=uuid.uuid4(),
+                            parameters={
+                                'properties': {
+                                    'principalId': f'{object_id}',
+                                    'principalType': 'User',
+                                    'roleDefinitionId': role_definition_id,
+                                }
+                            },
+                        )
+                    except azure.exceptions().ResourceExistsError as error:
+                        if 'StorageAccountAlreadyTaken' in error.message:
+                            # The error occurs when the storage account
+                            # name attempted to be created already exists
+                            # globally being used by other user.
+                            with ux_utils.print_exception_no_traceback():
+                                raise exceptions.StorageBucketCreateError(
+                                    'The storage account name '
+                                    f'{self.storage_account_name!r} is '
+                                    'already taken under other user\'s '
+                                    'subscripion. '
+                                    'Please try with another name.')
+                        if 'StorageAccountAlreadyExists' in error.message:
+                            # The error occurs when the storage account
+                            # name attempted to be created already exists
+                            # under the user's subscription.
+                            with ux_utils.print_exception_no_traceback():
+                                raise exceptions.StorageBucketCreateError(
+                                    'The storage account name '
+                                    f'{self.storage_account_name!r} '
+                                    'already exists under your '
+                                    'subsciption in another resource '
+                                    'group. Please try another name.')
```

Comment:
how about the expection type is still `azure.exceptions().ResourceExistsError`, but neither `StorageAccountAlreadyTaken` nor `StorageAccountAlreadyExists` exists in the message?

Reply:
Done https://github.com/skypilot-org/skypilot/pull/3032#discussion_r1646594533